src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -627,14 +731,27 @@ ov::npuw::LLMCompiledModel::LLMCompiledModel(const std::shared_ptr<ov::Model>& m
     const uint32_t min_response_len = align_to(m_cfg.get<::intel_npu::NPUW_LLM_MIN_RESPONSE_LEN>(), 64u);
 
     m_kvcache_desc = KVCacheDesc{max_prompt_len, max_prompt_len + min_response_len, 0u, seq_len_dim};
-    LOG_DEBUG("4. Make prefill model with static shapes");
+    LOG_DEBUG(++step_msg << ". Make prefill model with static shapes");
     reshape_to_static(prefill_model, m_kvcache_desc.max_prompt_size, m_kvcache_desc.max_prompt_size, axes);
-    LOG_DEBUG("5. Make kvcache model with static shapes");
+    LOG_DEBUG(++step_msg << ". Make kvcache model with static shapes");
     reshape_to_static(kvcache_model, 1u, m_kvcache_desc.total_size, axes);
+    if (tail_mm_model) {
+        LOG_DEBUG(++step_msg << ". Make tail matmul model with static shapes");
+        // We have only one input: output of Slice operation, and this output should be sliced
+        // to have 1 in 1st dim.
+        const auto& tmm_input = tail_mm_model->input(0);
+        ov::PartialShape new_shape;
+        const auto& partial_shape = tmm_input.get_partial_shape();
+        NPUW_ASSERT(partial_shape.size() == 3);
+        new_shape = partial_shape;
+        new_shape[axes.batch] = 1;
+        new_shape[1] = 1;
+        tail_mm_model->reshape(new_shape);
+    }
```

Comment:
probably, for the consistency of the abstraction level, you can move it to `reshape_tail_to_static` helper function and call it here. Otherwise there's too much details for the flow.

Reply:
Agree, thanks, fixed!